File: src/OrderPaper.Web/Models/OrderPaper.cs
using System;
using System.Collections.Generic;

namespace OrderPapers.Models
{
    public class OrderPaper
    {
        public int Id { get; set; }
        public DateTime SittingDay { get; set; }
        public string Status { get; set; }
        public string Number { get; set; }
        public string SittingHours { get; set; }
        public List<Section> Sections { get; set; }
    }
}

File: src/OrderPaper.Web/wwwroot/app/models/OrderPaper.ts
﻿import { Status }                                   from './constants';
import { Item, MotionItem, BillItem, ReportItem }   from './Items';
import { Section }                                  from './Section';
import { AppConstants }                             from '../settings/app.constants';

export class OrderPaper {
    Id: number;
    SittingDay: string;
    Status: string;
    Number: number;
    SittingHours: string;
    Sections: Array<Section>;
    Progress: string;
    PublishingProgress: Array<string>;
    constructor() {
        this.Sections = new Array<Section>();
        this.PublishingProgress = new Array<string>();
    }

    public containPreview = (): boolean => {
        return this.PublishingProgress.indexOf(AppConstants.PROGRESS_PREVIEW) >= 0;
    }

    public containWord = (): boolean => {
        return this.PublishingProgress.indexOf(AppConstants.PROGRESS_WORD) >= 0;
    }

    public containPublish = (): boolean => {
        return this.PublishingProgress.indexOf(AppConstants.PROGRESS_PUBLISH) >= 0;
    }

    public containPrint = (): boolean => {
        return this.PublishingProgress.indexOf(AppConstants.PROGRESS_PRINT) >= 0;
    }
}